File: logs.ts
export interface AgentLog {
  id: string;
  result_id: string;
  agent_name: string;
  action: string;
  input?: Record<string, any>;
  output?: Record<string, any>;
  error?: string;
  retry_count: number;
  timestamp: string;
}

export interface MetricsSnapshot {
  p50_latency_ms: number;
  p95_latency_ms: number;
  p99_latency_ms: number;
  avg_cost_usd: number;
  total_queries: number;
  success_rate: number;
}

File: logs.py
from typing import Optional, Any
from datetime import datetime
from pydantic import BaseModel, Field

class AgentLogCreate(BaseModel):
    result_id: str
    agent_name: str
    action: str
    input: Optional[dict[str, Any]] = None
    output: Optional[dict[str, Any]] = None
    error: Optional[str] = None
    retry_count: int = 0

class AgentLog(AgentLogCreate):
    id: str
    timestamp: datetime

class MetricsSnapshot(BaseModel):
    p50_latency_ms: float
    p95_latency_ms: float
    p99_latency_ms: float
    avg_cost_usd: float
    total_queries: int
    success_rate: float
